doc/dev/design/workflows/storage_devices_and_osds.rst

```diff
@@ -0,0 +1,304 @@
+
+.. prompt:: bash #
+
+  ceph orch device ls
+  ceph orch device ls json ( to get all the fields for each device )
+
+    Current Problems:
+        * Does not scale.
+
+**GUI**:
+    Operations:
+        * cluster.Inventory section:
+            The cluster.Inventory section presents a basic list of the devices in the cluster. It is a fixed list with only a few fields. Only the “ident light on” operation is possible although we do not know if it is possible or not until the operation is launched.
+
+    Problems
+        * Does not scale (depends of the orchestrator)
+        * Rigid user experience
+
+B. Proposed workflow:
+---------------------
+
+**CLI**:
+The current API is good enough, we only need to be sure that we have:
+
+    - all the attribute/health/operative state fields from each device
+    - fast response
+    - scalable
+
+**GUI**:
+The inventory should be able to be customized in order to show the desired fields of information for each device. Being customizable also the position of each field(column) and the sort order.
+
+The inventory should be filtered using any of the fields present in the list of devices.
+
+A customized inventory list together with the filter and sort order should be able to be stored for easy utilization. In this way we can provide a set of interesting predefined inventory lists. For example:
+
+    - Devices Available
+    - Devices more used (more average iops) (should be an alert/trigger)
+    - Devices biggers than n Gb
+
+The inventory should also provide a way to do directly operations over physical devices:
+
+    * Identify:  Start/stop to blink the identification light
+    * Create OSD: Create an OSD using this disk device if it is available.
+    * Remove OSD: Delete the OSD using this disk device.
+
+2. Add OSDs
+===========
+
+A. Current workflow
+--------------------
+
+**CLI**:
+We can specify specific devices or use a “drive group” specification to create OSD’s in different hosts. By default, the definition of the OSDs to create is “declarative“ unless you use the unmanaged parameter.
+
+.. prompt:: bash #
+
+  ceph orch daemon add osd <host>:device1,device2 [--unmanaged=true]
+  ceph orch apply osd -i <json_file/yaml_file> [--dry-run] [--unmanaged=true]*
+
+**GUI**:
+Implemented in the dashboard section “cluster.OSDs”.
+There is a button to create the OSDs, that presents a page dialog box to select the physical devices that are going to be part of the OSDs as main, WAL or Db devices.
+It Is very difficult to make a selection ( or to understand how to make the selection). This is even worse if your cluster has the same kind of devices, resulting in the weird thing that is not possible to create an OSD using only one storage device (because you cannot select it)
+The problem here is the UI has been designed to work with “drive groups” and not to work for the user. The “drive group” is an abstract concept that must be used only in the background. Users must not be aware of this concept.
```

Comment:
The current dashboard implementation is based on what `cephadm` provided. Usually, a UI should support a user to perform their task at hand with lesser effort, and not having to manually click each device individually in order to deploy OSDs on them is one such thing. We need to keep in mind that we want to make it easy for users managing possibly hundreds or thousands of disks with little effort. We consider the "select just a single device" the edge case here.

Reply:
The proposed workflow provides a design to make easy to work only with a single device and also to work with hundred of them.